resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/interceptors/encoding/MessageSanitizerContainerResponseFilter.java

```diff
@@ -0,0 +1,36 @@
+package org.jboss.resteasy.plugins.interceptors.encoding;
+
+import java.io.IOException;
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.ext.Provider;
+import org.apache.commons.lang.StringEscapeUtils;
+import org.jboss.resteasy.util.HttpResponseCodes;
+
+/**
+ * Thwart select XSS attack by escaping special chars in Exception message.
+ * <p/>
+ * User: rsearls
+ * Date: 9/16/16
+ */
+@Provider
+@Priority(Priorities.ENTITY_CODER)
+public class MessageSanitizerContainerResponseFilter implements ContainerResponseFilter {
+    @Override
+    public void filter(ContainerRequestContext requestContext,
+                       ContainerResponseContext responseContext) throws IOException {
+
+        if (HttpResponseCodes.SC_BAD_REQUEST == responseContext.getStatus()) {
+            Object entity = responseContext.getEntity();
+            if (entity != null && entity instanceof String) {
+                String tmpEscapedMsg = StringEscapeUtils.escapeXml((String) entity);
```

Comment:
I've a minor (performance) concern here. By looking at the StringEscapeUtils code, in case of xml escape, what ends up being done is a 5 chars replacement (double-quote, ampersand, less-than, greater-than, xml apostrophe). The entity string is scanned linearly. Then we use String#replaceAll for the slash replacement, which builds up a pattern, matches on it and replaces. How about we manually create the Pattern, provide a regexp matching all 6 chars and perform the replacement with that? Alternatively, how about we do the linear string scan in our code and replace the 6 chars in the same pass ?